### DISCUSSION TITLE
Out of shared memory (SQLSTATE 53200)

### DISCUSSION THREAD

**Discussion by jdesouza** [pgpartman/pg_partman#554] [General]
I've created a pre-built partitioned table with around 1K partitions.
I'm getting  "out of shared memory (SQLSTATE 53200)" when running my code.
Increased max_locks_per_transaction = 2048 but still getting error. Any advice?

**Comment by keithf4** [comment]
Sounds like you may be running out of shared_buffers possibly? Have you tried increasing that?
Maybe try doing an EXPLAIN (not EXPLAIN ANALYZE) on the query to get an idea of what it might be doing to cause high memory usage.

How many rows total in the entire partition set? 
How large is each child table expected to be (rowcount and byte size)?
If you're really hitting a large majority of the child tables in your queries (hence needing to increase max_locks_per_transaction), perhaps also consider lowering your partition interval to lower the number of partitions.
